app/models/manageiq/providers/redhat/inventory/collector/target_collection.rb

```diff
@@ -113,22 +113,30 @@ def vms
     v
   end
 
+  def select_vms(references_lst)
+    references_lst.select { |ref| ref.include?('vms') }
+  end
+
+  def select_templates(references_lst)
+    references_lst.select { |ref| ref.include?('templates') }
+  end
+
   def templates
-    temp = []
-    return temp if references(:templates).blank?
+    v = []
```

Comment:
v isn't fit here. the returned type is an array of templates, so 't' ?

Reply:
done